uefi/src/proto/hii/config_str.rs

```diff
@@ -232,13 +250,16 @@ impl FromStr for ConfigurationString {
                 }
                 _ => return Err(ParseError::BlockName),
             };
+            // The uefi specification declares `VALUE` as being a "number" of arbitrary length.
+            // And numbers have to be endianness-corrected. Thus, the bytes represented by a value's
+            // hex string of arbitrary length have to be reversed to account for that weird encoding.
             let value = match splitter.next() {
-                Some(("VALUE", Some(data))) => Self::parse_bytes_from_hex(data).collect(),
+                Some(("VALUE", Some(data))) => Self::parse_bytes_from_hex(data).rev().collect(),
                 _ => return Err(ParseError::BlockConfig),
             };
 
             while let Some(next) = splitter.peek() {
-                if next.0 == "OFFSET" {
+                if next.0 == "OFFSET" || next.0 == "GUID" {
```

Comment:
Would it make sense to have an enum or a module with constants for these values?